src/main/java/hudson/remoting/PingThread.java

```diff
@@ -134,16 +134,16 @@ private void ping() throws IOException, InterruptedException {
      *      and provide a fallback behaviour to be backward compatible with earlier version of remoting library.
      */
     @Deprecated
-    protected abstract void onDead();
+    protected void onDead() {
+        onDead(new RuntimeException("No cause"));
```

Comment:
What about a <code>@CheckForNull</code> method? seems to be a more convenient solution